pkg/cfn/builder/managed_launch_template_test.go

```diff
@@ -0,0 +1,154 @@
+package builder
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	. "github.com/benjamintf1/unmarshalledmatchers"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	"github.com/stretchr/testify/mock"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/testutils/mockprovider"
+	"github.com/weaveworks/goformation/v4"
+)
+
+type mngCase struct {
+	ng                *api.ManagedNodeGroup
+	resourcesFilename string
+	mockFetcherFn     func(*mockprovider.MockProvider)
+}
+
+var _ = Describe("ManagedNodeGroup builder", func() {
+	DescribeTable("Add resources", func(m *mngCase) {
+		clusterConfig := api.NewClusterConfig()
+		clusterConfig.Metadata.Name = "lt"
+		api.SetManagedNodeGroupDefaults(m.ng, clusterConfig.Metadata)
+		Expect(api.ValidateManagedNodeGroup(m.ng, 0)).To(Succeed())
+
+		provider := mockprovider.NewMockProvider()
+		if m.mockFetcherFn != nil {
+			m.mockFetcherFn(provider)
+		}
+
+		stack := NewManagedNodeGroup(clusterConfig, m.ng, NewLaunchTemplateFetcher(provider.MockEC2()), fmt.Sprintf("eksctl-%s", clusterConfig.Metadata.Name))
+		stack.UserDataMimeBoundary = "//"
+		err := stack.AddAllResources()
+		Expect(err).ToNot(HaveOccurred())
+		bytes, err := stack.RenderJSON()
+		Expect(err).ToNot(HaveOccurred())
+
+		template, err := goformation.ParseJSON(bytes)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(template).ToNot(BeNil())
+
+		actual, err := json.Marshal(template.Resources)
```

Comment:
Am I reading this correctly that the tests unmarshal and marshal and then compare the json? Why not match on the structs?

Reply:
That's right. The unmarshalling (`goformation.ParseJSON(bytes)`) and marshalling is actually testing goformation's ability to preserve isomorphism, to ensure no types or values are lost or are interpreted differently. 

>Why not match on the structs?

Matching on the structs will require more code than matching on the expected JSON output. This is also close to an end-to-end test.

Reply:
makes sense!